### ISSUE TITLE
includes target "X" which requires target "Vulkan-Headers" that is not in any export set.

### ISSUE THREAD

**Issue by axsaucedo** [KhronosGroup/Vulkan-Headers#157]
Currently I am looking to use Vulkan Headers directly from the repo, but it seems when I include with `target_link_library` I get the error includes target "X" which requires target "Vulkan-Headers" that is not in any export set. Is there a reason to not include an export of the vulkan-header targets? Or is there a different suggested way to add vulkan headers as part of a target that will be exported?

**Comment by axsaucedo** [comment]
It's worth mentioning that if I try to export the targets myself I get the following error:
```
CMake Error in external/Vulkan-Headers/CMakeLists.txt:
  Target "Vulkan-Headers" INTERFACE_INCLUDE_DIRECTORIES property contains
  path:

    "C:/.../external/Vulkan-Headers/include"

  which is prefixed in the source directory.
```


This is attempting to add a separate export target for vulkan-headers as follows:
```
install(TARGETS Vulkan-Headers EXPORT Vulkan-HeaderTargets
    LIBRARY DESTINATION lib
    ARCHIVE DESTINATION lib
    INCLUDES DESTINATION include)

install(EXPORT Vulkan-HeaderTargets
    FILE vulkanHeadersConfig.cmake
    NAMESPACE Vulkan::Headers::
    DESTINATION lib/cmake/vulkanheaders)
```

Or if I try to add it as part of my current export:

```
    install(TARGETS Xlib VulkanHeaders EXPORT XlibTargets
        LIBRARY DESTINATION lib
        ARCHIVE DESTINATION lib
        INCLUDES DESTINATION include)
```

**Comment by axsaucedo** [comment]
It seems that it's possible to circumvent the issue above by using this instead:
```
get_target_property(VULKAN_HEADERS_INCLUDES Vulkan-Headers INTERFACE_INCLUDE_DIRECTORIES)
target_include_directories(
    kompute PUBLIC
    $<BUILD_INTERFACE:${VULKAN_HEADERS_INCLUDES}>)
```
However I'm not sure if this is the proper way of doing this - are there best practices on how to include for install interfaces?

**Comment by loicmouton** [comment]
I came across the same issue while using FetchContent with this project and I think I found a solution but it would require to modify the CMakeLists.txt file.
The Vulkan-Headers cmake project does not follow the guidelines to create relocatable packages given in the CMake manual ([link](https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html#creating-relocatable-packages)).
The solution would be to replace:
`target_include_directories(Vulkan-Headers INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/include")`
by:
`target_include_directories(Vulkan-Headers INTERFACE
    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
)`
Additionally, it would be great to add a cmake cache variable that would control whether the targets defined in this project are added to the install rule. By doing so, projects using FetchContent to add this project to their build could easily set the value themselves. This is done in a lot of projects (namely SDL2 and fmt if I remember correctly).
